data/basics/overflow.go

```diff
@@ -152,3 +152,11 @@ func Muldiv(a uint64, b uint64, c uint64) (res uint64, overflow bool) {
 	quo, _ := bits.Div64(hi, lo, c)
 	return quo, false
 }
+
+// DivCeil provides `math.Ceil` semantics using integer division.  The technique
+// avoids slower floating point operations as suggested in https://stackoverflow.com/a/2745086.
+//
+// The method does _not_ check for divide-by-zero.
+func DivCeil[T constraints.Integer](numerator, denominator T) T {
+	return (numerator + denominator - 1) / denominator
+}
```

Comment:
I am unconvinced.
What about `DivCeil(-4, 3)` and `math.Ceil(-4/3)`

Reply:
I suppose technically it says "math.Ceil semantics" rather than math.Ceil results.  
I think (unless I'm wrong) this could be resolved by renaming the func or perhaps naming the parameters something that implies their non-negative-ness despite the declared type constraint or maybe enhancing the comment around what input it works for.
I know this code was around the repo before but now it's getting exported.

Reply:
Good point, this falls down for negative numbers. I think the best approach is to update the documentation to indicate this should only be used for positive numbers; supporting negative one is not currently necessary and seems too complex to justify right now.

Updated in 55fd6f76d50114f17340d0866871913b4712d748

Reply:
I think this is sufficient. The only reason we use `int` in some callers is Go's insistence that `int` is a good type for indexing arrays/slices. We expect them to be positive!